src/Disks/IO/CachedOnDiskReadBufferFromFile.cpp

```diff
@@ -1134,9 +1131,18 @@ bool CachedOnDiskReadBufferFromFile::nextImplStep()
             download_finished_time = "None";
 
         std::optional<size_t> object_size;
+        std::optional<size_t> impl_read_until_position;
+        std::optional<std::string> impl_read_stop_reason;
```

Comment:
empty string as for None instead of nullopt?

Reply:
I'd rather have nullopt to differentiate from the case when we did not get `stop_reason` from impl at all vs it was empty inside impl.